packages/components/src/button/index.tsx

```diff
@@ -174,7 +174,7 @@ export function UnforwardedButton(
 	const anchorProps: ComponentPropsWithoutRef< 'a' > =
 		Tag === 'a' ? { href, target } : {};
 
-	if ( disabled && isFocusable ) {
+	if ( ( disabled && isFocusable ) || isBusy ) {
```

Comment:
I disagree with this change. IMHO you might want to keep the button appearing busy, but still allow it to be non-disabled. If you really want to disable it, we do have the `disabled` prop that allows you to do it.